torch/_dynamo/symbolic_convert.py

```diff
@@ -1121,8 +1121,18 @@ def load_builtin(self, inst):
             assert is_builtin_constant(val)
             self.push(ConstantVariable.create(value=val))
 
-    def jump(self, inst):
-        self.instruction_pointer = self.indexof[inst.target]
+    def jump(self, inst, offset: Optional[int] = None):
+        # Jump to `inst.target` or jump by `offset`
+        if inst.target is not None:
+            assert offset is None
+            ip = self.indexof[inst.target]
+        else:
+            assert isinstance(self.instruction_pointer, int)
+            assert offset is not None
+            ip = self.instruction_pointer + offset
```

Comment:
Add an `assert sys.version_info >= (3, 10)`

This is the change I am worried about here ([source](https://docs.python.org/3/library/dis.html))
```
Changed in version 3.10: The argument of jump, exception handling and loop instructions is now the instruction offset rather than the byte offset.
```

In prior versions of python you would need to divide the offset by 2.  So if we ever hit this code in Python 3.8/3.9 it will result in some very weird bugs.

More fundamentally, why isn't `inst.target` being set properly?  Can we just fix the code so that it is defined and this isn't needed?

Reply:
Also, we will need to assert on the opcode.  There are functions like JUMP_ABSOLUTE/JUMP_BACKWARD that would use a different formula to compute the new `ip`.

Reply:
I reverted this change.